controllers/precache.go

```diff
@@ -298,18 +346,32 @@ func (r *ClusterGroupUpgradeReconciler) extractPrecachingSpecFromPolicies(
 			case utils.PolicyTypeClusterVersion:
 				image := object["spec"].(map[string]interface {
 				})["desiredUpdate"].(map[string]interface{})["image"]
-				if len(spec.PlatformImage) > 0 && spec.PlatformImage != image {
-					msg := fmt.Sprintf("Platform image must be set once, but %s and %s were given",
-						spec.PlatformImage, image)
-					meta.SetStatusCondition(&clusterGroupUpgrade.Status.Conditions, metav1.Condition{
-						Type:    utils.PrecacheSpecValidCondition,
-						Status:  metav1.ConditionFalse,
-						Reason:  "PlatformImageConflict",
-						Message: msg})
-					return *new(ranv1alpha1.PrecachingSpec), nil
+				if image != nil {
+					if len(spec.PlatformImage) > 0 && spec.PlatformImage != image {
+						msg := fmt.Sprintf("Platform image must be set once, but %s and %s were given",
+							spec.PlatformImage, image)
+						meta.SetStatusCondition(&clusterGroupUpgrade.Status.Conditions, metav1.Condition{
+							Type:    utils.PrecacheSpecValidCondition,
+							Status:  metav1.ConditionFalse,
+							Reason:  "PlatformImageConflict",
+							Message: msg})
+						return *new(ranv1alpha1.PrecachingSpec), nil
+					}
+					spec.PlatformImage = fmt.Sprintf("%s", image)
+				} else {
+					upstream := object["spec"].(map[string]interface {
+					})["upstream"].(string)
+					channel := object["spec"].(map[string]interface {
+					})["channel"].(string)
+					version := object["spec"].(map[string]interface {
+					})["desiredUpdate"].(map[string]interface{})["version"].(string)
+					image, err := r.getImageForVersionFromUpdateGraph(upstream, channel, version)
+					if err != nil {
+						return spec, err
```

Comment:
This might fail for variety of reasons:
- User left `upstream`, `channel` or `version` empty. 
- User made a typo in one of them
- The upstream server failed, or there is a temporary network issue
I wonder, would it be better to raise a condition (utils.PrecacheSpecValidCondition) and pass the error message as message?
This will give user a chance to fix the policy without digging in the log

Reply:
Good question, I do not know what is coming from ZTP.
Are you saying that it may not be present those fields?
What is the contract?

Reply:
If the answer is 'we can't assume anything' that's fine, i can check and if not present then return nil spec as it's been till now